clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java

```diff
@@ -546,6 +554,76 @@ public void testMetadataTopicExpiry() throws Exception {
         assertTrue(future.isDone(), "Request should be completed");
     }
 
+    @Test
+    public void testNodeLatencyStats() throws Exception {
```

Comment:
This is a new addition, the rest is just updates to reflect function signatures / callback changes.